jetty-client/src/main/java/org/eclipse/jetty/client/HttpDestination.java

```diff
@@ -104,23 +111,83 @@ public HttpDestination(HttpClient client, Origin origin)
         hostField = new HttpField(HttpHeader.HOST, host);
     }
 
+    public boolean stale()
+    {
+        try (Locker.Lock l = staleLock.lock())
+        {
+            boolean stale = this.stale;
+            if (!stale)
+                this.staleTs = 0;
```

Comment:
I don't think this is right. Should it not be `System.nanoTime()` instead of `0`?

Reply:
Good catch, fixed.